cmd/juju/model/destroy.go

```diff
@@ -258,13 +258,21 @@ func (c *destroyCommand) Run(ctx *cmd.Context) error {
 	}
 	defer func() { _ = api.Close() }()
 
+	modelStatuses, err := api.ModelStatus(names.NewModelTag(modelDetails.ModelUUID))
+	if err != nil {
+		return errors.Annotate(err, "getting model status")
+	}
+	modelStatus := modelStatuses[0]
```

Comment:
See the code in handlePersistentStorageError for how to check for ModelStatus errors